synced_bcf_reader.c

```diff
@@ -638,11 +638,12 @@ static void _reader_shift_buffer(bcf_sr_t *reader)
 {
     int i;
     for (i=2; i<=reader->nbuffer; i++)
-        if ( reader->buffer[i]->pos!=reader->buffer[1]->pos ) break;
+        if ( reader->buffer[i]->rid!=reader->buffer[1]->rid || reader->buffer[i]->pos!=reader->buffer[1]->pos ) break;
     if ( i<=reader->nbuffer )
     {
         // A record with a different position follows, swap it. Because of the reader's logic,
         // only one such line can be present.
+        assert( i==reader->nbuffer );
```

Comment:
Why not use the condition `i==reader->nbuffer` directly in the `if` statement above?

Reply:
To be able to test the assertion in case the assumption is violated?